src/node_stat_watcher.cc

```diff
@@ -111,6 +111,8 @@ void StatWatcher::Start(const FunctionCallbackInfo<Value>& args) {
   const bool persistent = args[1]->BooleanValue();
   const uint32_t interval = args[2]->Uint32Value();
 
+  if (uv_is_active(reinterpret_cast<uv_handle_t*>(wrap->watcher_)))
+    return;
```

Comment:
This is wrong when Start() is called with different arguments than last time.  This should work:
```c++
uv_fs_poll_stop(wrap->watcher_);
uv_fs_poll_start(wrap->watcher_, Callback, *path, interval);

// Safe, uv_ref/uv_unref are idempotent.
if (persistent)
  uv_ref(reinterpret_cast<uv_handle_t*>(wrap->watcher_));
else
  uv_unref(reinterpret_cast<uv_handle_t*>(wrap->watcher_));
```
And while we're here, this code really ought to check the return value of `uv_fs_poll_start()` (and `uv_fs_poll_stop()` too, although it can't fail in the current implementation.)

Reply:
Start() is not part of the external API, and no guarantees were made whether it works with different arguments or not. On the other hand, the proposed solution breaks compatibility.

Reply:
Then why don't you make it throw an exception?  Methods should not silently do the wrong thing, that's arguably worse than aborting.

> On the other hand, the proposed solution breaks compatibility.

In what way?

Reply:
The added test currently throws no exception. If I make it throw an exception it would be breaking compatibility.

Reply:
What do you mean?  There are no compatibility issues because it was aborting before.

Reply:
It was aborting in FSEventWrap, not StatWatcher. This was simply tacked on as a similar change that improves the current behavior in a similar way.

Reply:
I wouldn't use the word 'improves'.  StatWatcher already worked mostly like I proposed in my first comment and that would be an acceptable change for FSEventWrap too, as would throwing an exception.  Silently ignoring changes is the worst possible solution however.